File: projects/m1/019-average/solutions/alessiacaporaso/average.js
// ask value at user
let userValue;
let collectionValue = [];
while(userValue !== 0){
  userValue = parseInt(prompt('Please, enter the value'))
  collectionValue.push(userValue)
}
// sum element of array
let sumElement = 0;
for(const value of collectionValue){
  sumElement += value;
}
// calc average of value
const calcAverage = sumElement / (collectionValue.length - 1)
// display message, if the first value is 0 display error else display average
let message;
if(collectionValue <1 && userValue  === 0){
  message = 'Error, the first value entered cannot be zero'  
} else {
   message = `The average of value is ${calcAverage}` 
}
alert(message)

File: projects/m1/019-average/solutions/RoccoFigliamonti/average.py
"""# Average

In this exercise you will create a program that computes the average of a collection of values entered by the user.
The user will enter 0 as a sentinel value to indicate that no further values will be provided.
Your program should display an appropriate error message if the first value entered by the user is 0.

Hint: Because the 0 marks the end of the input it should not be included in the average.
"""

num=[]
while True:
    num.append(int(input("Please insert a number: ")))
    if num[0] == 0:
        print('error')
        break
    elif num[-1] == 0:
        result = sum(num) / (len(num)-1)
        print(f'The Average of the list is: {round(result,2)}')
        break

